src/playsim/p_enemy.cpp

```diff
@@ -944,16 +945,21 @@ void P_NewChaseDir(AActor * actor)
 		if (actor->flags3 & MF3_AVOIDMELEE)
 		{
 			bool ismeleeattacker = false;
-			double dist = actor->Distance2D(target);
+
+			double distance_squared = actor->Distance2DSquared(target);
 			if (target->player == NULL)
 			{
-				ismeleeattacker = (target->MissileState == NULL && dist < (target->meleerange + target->radius)*2);
+				//in its own variable because otherwise it looks bad
+				double meele_squared = (target->meleerange + target->radius) * 2 * (target->meleerange + target->radius) * 2;
+				ismeleeattacker = (target->MissileState == NULL && distance_squared < meele_squared);
```

Comment:
it's better safe than sorry if you do this:

```suggestion
				double meele = (target->meleerange + target->radius)*2;
				ismeleeattacker = (target->MissileState == NULL && (melee > 0 && distance_squared < meele*melee));
```